src/main/java/graphql/util/MemoizedSupplier.java

```diff
@@ -0,0 +1,28 @@
+package graphql.util;
+
+import java.util.function.Supplier;
+
+import static graphql.Assert.assertNotNull;
+
+class MemoizedSupplier<T> implements Supplier<T> {
```

Comment:
@andimarek @felipe-gdr  - thoughts on this code pattern??

My aim is to not lock - so possible that IF a DataFetchingEnv some how crossed threads then its possible that 2 reads cause 2 supplier reads.  But its lock free and I think MOST cases will be read on the thread that has the DF

To be pathological, you would need to read it on thread A and then read it again on thread B.